templates/project-base.html

```diff
@@ -133,6 +133,30 @@ <h1 id="project-title">
   </div> <!-- /col -->
 </div> <!-- /row -->
 
+{% block scripts %}
+    <script>
+        function show_import_project_modal(id, callback) {
+            var m = $('#import_project_modal');
+            $("#import_loading_spinner").hide();
+            m.modal();
+        }
+
+        function fillJSONContent(file) {
+            filecontents = $('#{{import_project_form.file.auto_id}}').prop('files')[0];
+            var reader = new FileReader();
+            reader.readAsText(filecontents);
+            reader.onload = function (e) {
+                $('#{{import_project_form.json_content.auto_id}}').val(e.target.result);
+            };
+        }
+
+        function show_spinner() {
+            $("#import_loading_spinner").show();
+        }
+    </script>
+{% endblock %}
+{% include "controls/import_project_modal.html" %}
```

Comment:
This code is being duplicated between this file and the import_project_modal (which is included in the last line here), and ideally also with the component import template (where it shouldn't be removed), so that's looking like a lot of JS duplication. This makes me think the JS should be separated into a template somehow. Or maybe these entire modals should be separated into their own dedicated "import_modal" template.

Reply:
Ended up I needed to send an import_name so I made some adjustments with the js function fillProjectJSONContent